src/app/clients/add/AddClientForm.tsx

```diff
@@ -0,0 +1,194 @@
+import NumberChildrenCard from "@/app/clients/add/formSections/NumberChildrenCard";
+import DietaryRequirementCard from "@/app/clients/add/formSections/DietaryRequirementCard";
+import FeminineProductCard from "@/app/clients/add/formSections/FeminineProductCard";
+import BabyProductCard from "@/app/clients/add/formSections/BabyProductCard";
+import PetFoodCard from "@/app/clients/add/formSections/PetFoodCard";
+import OtherItemsCard from "@/app/clients/add/formSections/OtherItemsCard";
+import DeliveryInstructionsCard from "@/app/clients/add/formSections/DeliveryInstructionsCard";
+import ExtraInformationCard from "@/app/clients/add/formSections/ExtraInformationCard";
+import {
+    ClientDatabaseRecord,
+    insertClient,
+    insertFamily,
+} from "@/app/clients/add/databaseFunctions";
+import attentionFlagCard from "@/app/clients/add/formSections/AttentionFlagCard";
+import SignpostingCallCard from "@/app/clients/add/formSections/SignpostingCallCard";
+
+const initialFields: Fields = {
+    fullName: "",
+    phoneNumber: "",
+    addressLine1: "",
+    addressLine2: "",
+    addressTown: "",
+    addressCounty: "",
+    addressPostcode: "",
+    adults: [
+        { personType: "adult", quantity: 0 },
+        { personType: "male", quantity: 0 },
+        { personType: "female", quantity: 0 },
+    ],
+    numberChildren: 0,
+    children: [],
+    dietaryRequirements: {},
+    feminineProducts: {},
+    babyProducts: null,
+    nappySize: "",
+    petFood: {},
+    otherItems: {},
+    deliveryInstructions: "",
+    extraInformation: "",
+    attentionFlag: false,
+    signpostingCall: false,
+};
+
+const initialFormErrors: FormErrors = {
+    fullName: Error.initial,
+    phoneNumber: Error.none,
+    addressLine1: Error.initial,
+    addressPostcode: Error.initial,
+    adults: Error.initial,
+    numberChildren: Error.initial,
+    nappySize: Error.none,
+};
+
+const AddClientForm: React.FC = () => {
+    const router = useRouter();
+    const [fields, setFields] = useState<Fields>(initialFields);
+    const [formErrors, setFormErrors] = useState<FormErrors>(initialFormErrors);
+    const [submitError, setSubmitError] = useState<Error>(Error.none);
+
+    useEffect(() => {
+        if (fields.numberChildren <= fields.children.length) {
+            fieldSetter("children", fields.children.slice(0, fields.numberChildren));
+            return;
+        }
+
+        const extraChildren: Person[] = Array(fields.numberChildren - fields.children.length)
+            .fill(0)
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            .map((_item) => {
+                return {
+                    personType: "child",
+                    age: null,
+                };
+            });
+        fieldSetter("children", [...fields.children, ...extraChildren]);
+    }, [fields.numberChildren]); // eslint-disable-line react-hooks/exhaustive-deps
+
```

Comment:
Same as above, does this `// eslint-disable-line react-hooks/exhaustive-deps` need to be here?

Reply:
Yep, this rule is not disabled globally (and probably shouldn't be?) but in this case we only want the `useEffect `to depend on `fields.numberChildren` even though it uses the other `useStates `as functions.